src/garage/torch/policies/discrete_qf_derived_policy.py

```diff
@@ -0,0 +1,68 @@
+"""A Discrete QFunction-derived policy.
+
+This policy chooses the action that yields to the largest Q-value.
+"""
+import numpy as np
+import torch
+
+from garage.torch.policies.policy import Policy
+
+
+class DiscreteQFDerivedPolicy(Policy):
+    """Policy that derives its actions from a learned Q function.
+
+    The action returned is the one that yields the highest Q value for
+    a given state, as determined by the supplied Q function.
+
+    Args:
+        qf (object): Q network.
+        env_spec (EnvSpec): Environment specification.
+        name (str): Name of this policy.
+    """
+
+    def __init__(self, qf, env_spec, name='DiscreteQFDerivedPolicy'):
+        super().__init__(env_spec, name)
+        self._qf = qf
+
+    # pylint: disable=arguments-differ
+    def forward(self, observations):
+        """Get actions corresponding to a batch of observations.
+
+        Args:
+            observations(torch.Tensor): Batch of observations of shape
+                :math:`(N, O)`. Observations should be flattened even
+                if they are images as the underlying Q network handles
+                unflattening.
+
+        Returns:
+            torch.Tensor: Batch of actions of shape :math:`(N, A)`
+        """
+        qs = self._qf(observations)
+        return torch.argmax(qs, dim=1)
```

Comment:
do you need to differentiate through this? if not, it could just be in garage.np.policies

Reply:
No, but I kept it under `torch.policies` because it subclasses `garage.torch.policy` to access the default methods defined there.

Reply:
hmm examples?

Reply:
in any case, probably best addressed when we have a proper PolicyWrapper class

Reply:
`policy.get_param_values()` and `policy.set_param_values()`, both of which are used by the runner/sampler.

Reply:
good to know -- thanks!